index.js

```diff
@@ -22,6 +22,9 @@ async function handleRequest(request) {
   else if(password.match('[a-z]') === null) {
     badPasswordMessage = 'Password must contain at least 1 lowercase character';
   }
+  else if (request.cf.postalCode && !password.includes(request.cf.postalCode)) {
+    badPasswordMessage = `Password must contain your postal code (${request.cf.postalCode})`
+  }
```

Comment:
```suggestion
  else if (request.cf.postalCode && !password.match(new RegExp(request.cf.postalCode.slice(0, 3) + '\d{2,4}')) {
    badPasswordMessage = `Password must contain your postal code (${request.cf.postalCode.slice(0, 3)}**)`
  }
```

As people pointed out here - postal codes does not match exact location. Yet they still match the country and city, so why not check first 3 digits instead?

Reply:
In US such accuracy would give down to state, in Japan/Russia/Ukraine - down to city. In EU it still completely misses though